components/builder-jobsrv/src/data_store.rs

```diff
@@ -43,10 +51,100 @@ use crate::error::{Error,
 
 mod test;
 
+// Holds the intermediate state of the 'jobs'
```

Comment:
Should we be using the builder-db models for these types? Perhaps even get rid of most/all of this raw sql by leveraging the diesel APIs.

Reply:
Looking at the functions that this raw sql is calling, I think the vast majority of them can be discarded in favor of using the diesel models directly or adding functions to the model that reproduce what the function does. Examples are:

```
CREATE OR REPLACE FUNCTION sync_jobs_v2() RETURNS SETOF jobs
    LANGUAGE sql STABLE
    AS $$
  SELECT * FROM jobs WHERE (scheduler_sync = false) OR (sync_count > 0);
$$;

CREATE OR REPLACE FUNCTION find_group_project_v1(gid bigint, name text) RETURNS SETOF group_projects
    LANGUAGE plpgsql STABLE
    AS $$
BEGIN
  RETURN QUERY SELECT * FROM group_projects WHERE owner_id = gid AND project_name = name;
  RETURN;
END
$$;
```

I do see a couple functions where we would likely need to keep the function but it looks like diesel provides a way to map to funcrions. See https://github.com/habitat-sh/builder/blob/master/components/builder-db/src/functions/jobs.rs for some examples. These seem to be clear straight forward in cases where the function returns `VOID` or a single value. However, I'm not sure how to set it up for functions that return a dataset. The only function I see that could be tough to map (with my limited diesel knowledge) is `next_pending_job_v2`.